python/ql/src/experimental/Functions/NamingConventionsFunctions.ql

```diff
@@ -0,0 +1,17 @@
+/**
+ * @name Misnamed function
+ * @description A function name that begins with an uppercase letter decreases readability.
+ * @kind problem
+ * @problem.severity recommendation
+ * @precision medium
+ * @id python/misnamed-function
+ * @tags maintainability
+ */
+
+import python
+
+from Function f
+where
+  f.inSource() and
+  not f.getName().substring(0, 1).toLowerCase() = f.getName().substring(0, 1)
+select f, "Function names should start in lowercase."
```

Comment:
```suggestion
from Function f, string first_char
where
  f.inSource() and
  first_char = f.getName().prefix(1) and
  not first_char = first_char.toLowerCase()
select f, "Function names should start in lowercase."
```
I would suggest rewriting the query as follows, to eliminate the repeated `substring`, and to (hopefully) make the intent clearer. A similar comment applies to the other query.

Reply:
This has been changed to the method you have suggested here. With regards to the first suggestion you made, I modified the queries so that it only enters the select statement if the class/function is the only one in its file that has the incorrect case. On lgtm this reduced the number of results from ~15 per project to ~7 per project, eliminating many false positives I believe. Let me know if you think that makes sense or if a different approach would be better. Thanks again!